packages/backend/src/managers/llama-stack/llamaStackManager.ts

```diff
@@ -0,0 +1,251 @@
+  }
+
+  async createLlamaStackContainer(
+    connection: ContainerProviderConnection,
+    labels: { [p: string]: string },
+  ): Promise<LlamaStackContainerInfo> {
+    const image = llama_stack_images.default;
+    const pullingTask = this.taskRegistry.createTask(`Pulling ${image}.`, 'loading', labels);
+    const imageInfo = await getImageInfo(connection, image, () => {})
+      .catch((err: unknown) => {
+        pullingTask.state = 'error';
+        pullingTask.progress = undefined;
+        pullingTask.error = `Something went wrong while pulling ${image}: ${String(err)}`;
+        throw err;
+      })
+      .then(imageInfo => {
+        pullingTask.state = 'success';
+        pullingTask.progress = undefined;
+        return imageInfo;
+      })
+      .finally(() => {
+        this.taskRegistry.updateTask(pullingTask);
+      });
+
+    const folder = await this.getLlamaStackContainerFolder();
+
+    const containerTask = this.taskRegistry.createTask('Starting Llama Stack container', 'loading', labels);
+    const aiLabApiPort = this.configurationRegistry.getExtensionConfiguration().apiPort;
+    const llamaStackApiPort = await getFreeRandomPort('0.0.0.0');
+    const createContainerOptions: ContainerCreateOptions = {
+      Image: imageInfo.Id,
+      name: getRandomName('llama-stack'),
+      Labels: {
+        [LLAMA_STACK_CONTAINER_LABEL]: image,
+        [LLAMA_STACK_API_PORT_LABEL]: `${llamaStackApiPort}`,
+      },
+      HostConfig: {
+        AutoRemove: true,
+        SecurityOpt: [DISABLE_SELINUX_LABEL_SECURITY_OPTION],
+        Mounts: [
+          {
+            Target: '/app/.llama',
+            Source: path.join(folder, '.llama'),
+            Type: 'bind',
+          },
+        ],
+        UsernsMode: 'keep-id:uid=0,gid=0',
+        PortBindings: {
+          '8321/tcp': [
+            {
+              HostPort: `${llamaStackApiPort}`,
+            },
+          ],
+        },
+      },
+      Env: [`PODMAN_AI_LAB_URL=http://host.docker.internal:${aiLabApiPort}`],
+      OpenStdin: true,
+      start: true,
+    };
+    try {
+      const { id } = await containerEngine.createContainer(imageInfo.engineId, createContainerOptions);
+      // update the task
+      containerTask.state = 'success';
+      containerTask.progress = undefined;
+      return {
+        containerId: id,
+        port: llamaStackApiPort,
+      };
+    } catch (err: unknown) {
+      containerTask.state = 'error';
+      containerTask.progress = undefined;
+      containerTask.error = `Something went wrong while creating container: ${String(err)}`;
+      throw err;
+    } finally {
+      this.taskRegistry.updateTask(containerTask);
+    }
+  }
+
+  private async getLlamaStackContainerFolder(): Promise<string> {
+    const llamaStackPath = path.join(this.appUserDirectory, 'llama-stack', 'container');
+    await fs.mkdir(llamaStackPath, { recursive: true });
+    await fs.mkdir(path.join(llamaStackPath, '.llama'), { recursive: true });
```

Comment:
```suggestion
    await fs.mkdir(path.join(llamaStackPath, '.llama'), { recursive: true });
```

not needed to make the parent first, as we use recursive